server/src/tenant/entity-resolver/utils/pg-graphql-query-builder.util.ts

```diff
@@ -0,0 +1,128 @@
+import { GraphQLResolveInfo } from 'graphql';
+import graphqlFields from 'graphql-fields';
+import { v4 as uuidv4 } from 'uuid';
+
+import { pascalCase } from 'src/utils/pascal-case';
+
+import { stringifyWithoutKeyQuote } from './stringify-without-key-quote.util';
+import { convertFieldsToGraphQL } from './convert-fields-to-graphql.util';
+
+type Command = 'findMany' | 'findOne' | 'createMany' | 'updateOne';
+
+type CommandArgs = {
+  findMany: null;
+  findOne: { id: string };
+  createMany: { data: any[] };
+  updateOne: { id: string; data: any };
+};
+
+export interface PGGraphQLQueryBuilderOptions {
+  entityName: string;
+  tableName: string;
+  info: GraphQLResolveInfo;
+  fieldAliases: Record<string, string>;
+}
+
+export class PGGraphQLQueryBuilder {
+  private options: PGGraphQLQueryBuilderOptions;
+  private command: Command;
+  private commandArgs: any;
+
+  constructor(options: PGGraphQLQueryBuilderOptions) {
+    this.options = options;
+  }
+
+  private getFields(): string {
+    const fields = graphqlFields(this.options.info);
+
+    return convertFieldsToGraphQL(fields, this.options.fieldAliases);
+  }
+
+  // Define command setters
+  findMany() {
```

Comment:
I understand your concern on that, I don't think we need to handle that case.
I go this way at first because I was handling args, and other stuff in the chaining mechanism.
But in fact I should just remove the `build` and directly return the results from the commands.

Reply:
I agree 👍